javalib/src/main/scala/java/util/DoubleSummaryStatistics.scala

```diff
@@ -0,0 +1,63 @@
+package java.util
+
+import java.{lang => jl}
+
+class DoubleSummaryStatistics() {
+  var _count: Long = 0L
+  var _min: Double = jl.Double.POSITIVE_INFINITY
+  var _max: Double = jl.Double.NEGATIVE_INFINITY
+  var _sum: Double = 0.0
```

Comment:
Is there some reason why field names start with `_`? Also they could be made private
```suggestion
  private var count: Long = 0L
  private var min: Double = jl.Double.POSITIVE_INFINITY
  private var max: Double = jl.Double.NEGATIVE_INFINITY
  private var sum: Double = 0.0
```

In fact, there could also be defined within the primary constructor, but it's not crucial

Reply:
The same idiom is used in all three of the SummaryStatistics files, so it is worth my moving it from
"get the silly thing correct" to something presentable.   I believe at the time I was concerned at
a coarser granularity with getting `Collectors` to work.

You mention several things.  I will have to take some time to consider & unpack.

##### re: "private"  
Yes, they should be "private" no matter where they are declared. Good catch.

###### re: underbar in names.  
I think at the time I wrote the first of these I was having 
  problems with conflicts when I had variables and arguments with the same name: understandable.
```
 def this(count: Long, min: Double, max: Double, sum: Double) = {
    this()
    _count = count
    _min = min
    _max = max
    _sum = sum
  }
```
I fell back to other examples in the code base which use "_" to indicate global variables.  
Looking at this, I could try an idiom such as ` this.count = count`. 

If that worked, it would eliminate some astonishment and WT? for reviewers.

###### re: variables in primary constructors.

* I know how one declares  a `var` in a primary  constructor, I do not know how one gives it
   an initial value.  "count" and "sum" will get initialized to the proper 0.0. "min" and "max"
   need values different than the default 0.0.

* Probably 99.9% of the time, the no-args constructor will be used.  My thinking was to reduce
   overhead by having that as the primary constructor.  This imposes some cost on the 4 arg
   constructor in that its `this()` sets variables and then it has to overwrite them. I think the
   amortized cost having the zero-arg constructor will be lower, but I have no concrete data.

Is there something more idiomatic or "normal" to do here?   I am open to suggestions/improvement,
especially since it is  "times three"

Reply:
Later, re: _ in variable name.

Whilst working on the Enum recommendation below, I came across the place in `Collector.scala` where
I was having a problem with the Scala 3 (?) compiler not liking argument names and `def` names being 
the same.  Is there a  more idiomatic solution than underbar/underscore?

Reply:
In Scala 3 having the following class is not allowed
```
class Foo(someValue: Int) {
  def someValue(): Int =  someValue
}
```
Actually usage of underscore `_someValue` or other similar form is probably the cleanest solution and the most idiomatic solution.